h/search/core.py

```diff
@@ -58,6 +58,11 @@ def run(self, params):
 
         return SearchResult(total, annotation_ids, reply_ids, aggregations)
 
+    def clear(self):
+        """Clear search filters, aggregators, and matchers."""
+        self.builder = query.Builder()
+        self.reply_builder = query.Builder()
```

Comment:
This is nice - I had been thinking that, since the filters, aggregators and matchers are actually stored and used in `Builder` not in `Search`, you would have to add a `clear()` method to `Builder` and add a `clear()` method to `Search` that calls it. And then you'd be implicitly relying on the fact that `Builder` doesn't actually do anything with an appended filter, matcher or aggregator until `build()` is called so that it's safe to clear them. So just replacing the entire `Builder` objects is much better :+1: